apps/prairielearn/src/pages/instructorFileEditor/instructorFileEditor.js

```diff
@@ -466,616 +453,148 @@ async function writeEdit(fileEdit) {
-        env: gitEnv,
-        on_success: _push,
-        on_error: _cleanupAfterWrite,
-        no_job_sequence_update: true,
-      };
-      serverJobs.spawnJob(jobOptions);
-    };
+      }
 
-    const _push = function () {
-      debug(`${job_sequence_id}: _push`);
-      const jobOptions = {
-        course_id: options.course_id,
-        user_id: options.user_id,
-        authn_user_id: options.authn_user_id,
-        job_sequence_id: job_sequence_id,
-        type: 'git_push',
-        description: 'Push to remote',
-        command: 'git',
-        arguments: ['push'],
-        working_directory: fileEdit.coursePath,
-        env: gitEnv,
-        on_success: _didSave,
-        on_error: _cleanupAfterCommit,
-        no_job_sequence_update: true,
-      };
-      serverJobs.spawnJob(jobOptions);
-    };
+      const fullPath = path.join(fileEdit.coursePath, fileEdit.dirName, fileEdit.fileName);
+      const contents = await fs.readFile(fullPath, 'utf8');
+      fileEdit.diskHash = getHash(b64Util.b64EncodeUnicode(contents));
+      if (fileEdit.origHash !== fileEdit.diskHash) {
+        throw new Error(`Another user made changes to the file you were editing.`);
+      }
 
-    const _didSave = () => {
-      debug(`${job_sequence_id}: _didSave`);
-      const jobOptions = {
-        course_id: options.course_id,
-        user_id: options.user_id,
-        authn_user_id: options.authn_user_id,
-        type: 'did_save',
-        description: 'Mark edit as saved',
-        job_sequence_id: job_sequence_id,
-        on_success: _unlock,
-        on_error: _cleanup,
-        no_job_sequence_update: true,
-      };
-      serverJobs.createJob(jobOptions, (err, job) => {
-        if (ERR(err, (err) => logger.info(err))) {
-          _finishWithFailure();
-          return;
-        }
+      await fs.writeFile(fullPath, b64Util.b64DecodeUnicode(fileEdit.editContents), 'utf8');
+      job.verbose(`Wrote changed to ${fullPath}`);
 
-        debug('Mark edit as saved');
-        updateDidSave(fileEdit, (err) => {
-          if (err) {
-            job.fail(err);
-          } else {
-            job.verbose('Marked edit as saved');
-            job.succeed();
-          }
-        });
-      });
-    };
-
-    const _unlock = () => {
-      debug(`${job_sequence_id}: _unlock`);
-      const jobOptions = {
-        course_id: options.course_id,
-        user_id: options.user_id,
-        authn_user_id: options.authn_user_id,
-        type: 'unlock',
-        description: 'Unlock',
-        job_sequence_id: job_sequence_id,
-        on_success: jobSequenceHasFailed ? _finishWithFailure : _getEndCommitHash,
-        on_error: _finishWithFailure,
-        no_job_sequence_update: true,
-      };
-      serverJobs.createJob(jobOptions, (err, job) => {
-        if (ERR(err, (err) => logger.info(err))) {
-          _finishWithFailure();
-          return;
+      if (config.fileEditorUseGit) {
+        try {
+          await job.exec('git', ['reset'], {
```

Comment:
Remind me why we are doing `git reset` here? Is this just a general "reset to get everything back to an unmodified state"? If so, would `git reset --hard` be more appropriate?

Reply:
I'm fine to leave things as they are, however, on the principle that it's working and we shouldn't break working code 😄

Reply:
Yeah, I'm going to avoid touching this for the moment.

FWIW, we have very very similar logic both here and in `lib/editors.js`, and that's a smell to me. I'd like to ideally rewrite `instructorFileEditor` to reuse our other editing machinery, which does more robust resetting of state:

https://github.com/PrairieLearn/PrairieLearn/blob/cdb0f21098ca80014ccb926b8d97812f0e185307/apps/prairielearn/src/lib/editors.js#L58-L64

I opened #8177 to track this work.